app/components/subject-viewer.cjsx

```diff
@@ -44,6 +44,11 @@ module.exports = React.createClass
     frameDimensions: {}
     inFlipbookMode: @props.allowFlipbook
 
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.annotation.task isnt @props.annotation?.task # task changed!
+      @props.classification._hidePreviousMarks = false
```

Comment:
I don't think this should be in subject-viewer, but I couldn't find another way determine when the task changes.